workflow/controller/operator.go

```diff
@@ -1691,6 +1691,9 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 					woc.computeMetrics(processedTmpl.Metrics.Prometheus, localScope, realTimeScope, false)
 				}
 			}
+			if processedTmpl.Synchronization != nil {
```

Comment:
Node status will be calculated here for retry nodes. The lock has to be released if the node is fulfilled